src/lightning/pytorch/callbacks/throughput_monitor.py

```diff
@@ -0,0 +1,127 @@
+# Copyright The Lightning AI team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import time
+from typing import TYPE_CHECKING, Any, Callable, Optional, Union
+
+import torch
+
+from lightning.fabric.plugins import Precision
+from lightning.fabric.utilities.throughput_monitor import _get_flops_available, _ThroughputMonitor
+from lightning.fabric.utilities.throughput_monitor import (
+    _plugin_to_compute_dtype as fabric_plugin_to_compute_dtype,
+)
+from lightning.pytorch.callbacks import Callback
+from lightning.pytorch.plugins import (
+    DoublePrecisionPlugin,
+    FSDPPrecisionPlugin,
+    MixedPrecisionPlugin,
+    PrecisionPlugin,
+    TransformerEnginePrecisionPlugin,
+)
+from lightning.pytorch.plugins.precision.bitsandbytes import BitsandbytesPrecisionPlugin
+from lightning.pytorch.plugins.precision.deepspeed import DeepSpeedPrecisionPlugin
+from lightning.pytorch.plugins.precision.half import HalfPrecisionPlugin
+from lightning.pytorch.plugins.precision.xla import XLAPrecisionPlugin
+from lightning.pytorch.trainer.states import TrainerFn
+from lightning.pytorch.utilities.rank_zero import rank_zero_info, rank_zero_only
+
+if TYPE_CHECKING:
+    from lightning.pytorch import LightningModule, Trainer
+
+
+class ThroughputMonitor(Callback):
+    r"""Tracks and logs throughput with the :class:`lightning.fabric.utilities.throughput_monitor.._ThroughputMonitor`.
+
+    Only support for ``trainer.fit`` is currently implemented.
+
+    Args:
+        length_fn: A function to compute the number of items in a sample given a batch.
+        batch_size_fn: A function to compute the number of samples given a batch.
+        \**kwargs: See available parameters in
+            :class:`lightning.fabric.utilities.throughput_monitor.._ThroughputMonitor`
+
+    """
+
+    def __init__(self, length_fn: Callable[[Any], int], batch_size_fn: Callable[[Any], int], **kwargs: Any) -> None:
+        super().__init__()
+        self.kwargs = kwargs
+        self.length_fn = length_fn
+        self.batch_size_fn = batch_size_fn
+        self._monitor: Optional[_ThroughputMonitor] = None
+        self._train_t0 = 0.0
+        self._total_lengths = 0
+
+    def setup(self, trainer: "Trainer", pl_module: "LightningModule", stage: str) -> None:
+        if stage is not TrainerFn.FITTING:
+            # TODO: this could use a monitor per stage
+            raise NotImplementedError(f"`trainer.{stage}()` is not supported")
```

Comment:
Adding support for the other stages shouldn't be difficult. Although I'm not sure that there's interest for it. Leave it for a follow-up or do it here?